### PR TITLE
MODINVSTOR-301 Deletion of a Resource type requires a check to determ…

### PR DESCRIPTION
**Pull request by Anatolii-Starkov**
-added the foreign key from instance table to the instance_type table
-added test

### CODE REVIEW

File: descriptors/ModuleDescriptor-template.json [hunk 1]
```diff
@@ -66,7 +66,7 @@
     },
     {
       "id": "instance-storage",
-      "version": "7.1",
+      "version": "8.0",
```
Comment (marcjohnson-kint) [inline descriptors/ModuleDescriptor-template.json]: Why is this a breaking compatibility change?

Comment (Anatolii-Starkov) [reply descriptors/ModuleDescriptor-template.json]: Probably yes

Comment (marcjohnson-kint) [reply descriptors/ModuleDescriptor-template.json]: > Probably yes

Yes, to causing confusion?

How could I have described that differently? Are you aware of the difference in FOLIO between interface and implementation compatibility?

Comment (Anatolii-Starkov) [reply descriptors/ModuleDescriptor-template.json]: Sorry Marc I didn't catch it at once. I will fix it. It should be 7.2 for the instance-storage and 0.3 for the instance-storage-batch

File: ramls/instance-storage-batch.raml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 #%RAML 1.0
 title: Inventory Storage Batch API
-version: v0.2
+version: v1.0
```
Comment (Igor-Gorchakov) [inline ramls/instance-storage-batch.raml]: Why the Interface version has increment if batch-API has not changed ?

File: descriptors/ModuleDescriptor-template.json (hunk 1 above)
Comment (marcjohnson-kint) [reply descriptors/ModuleDescriptor-template.json]: Why does the interface version need to change in any way? As far as I can tell, the interface hasn't been changed in any way.

What has changed is that the implementation is choosing to reject instances where the resource type reference does not refer to a resource type reference record?

Comment (Anatolii-Starkov) [reply descriptors/ModuleDescriptor-template.json]: Is it fine now? If no please provide your input how it should look like

**Comment by Anatolii-Starkov** [comment]
Thanks Marc for you explanation. I have refreshed my knowledge about versions on FOLIO 
(https://dev.folio.org/guidelines/contributing/#version-numbers)
So, I reverted versions of API 
instance-storage from 7.3 to 7.2 
instance-storage-batch from 0.3 to 0.2
and increased version of implementation from 16.1.0-SNAPSHOT to 17.0.0-SNAPSHOT

File: src/test/java/org/folio/rest/api/HoldingsStorageTest.java [hunk 3]
```diff
@@ -574,36 +569,7 @@ private JsonObject uprooted(UUID id) {
     tags.add("test-tag");
 
     return createInstanceRequest(id, "TEST", "Uprooted",
-      identifiers, contributors, UUID_TEXT, tags);
-  }
-
-  private JsonObject temeraire(UUID id) {
```
Comment (marcjohnson-kint) [inline src/test/java/org/folio/rest/api/HoldingsStorageTest.java]: Why has this been removed?

Comment (Anatolii-Starkov) [reply src/test/java/org/folio/rest/api/HoldingsStorageTest.java]: It isn't used

Comment (marcjohnson-kint) [reply src/test/java/org/folio/rest/api/HoldingsStorageTest.java]: Ok, thanks. In future, please try to minimize the amount of extraneous changes within a pull request. 

Please feel free to issue preparatory pull requests that perform this kind of cleanup only.

**Pull request merged**